### PR TITLE
Adds a user to `envoy-windows` image

### PR DESCRIPTION
**Pull request by davinci26**
Commit Message:

There are multiple scenarios where users want to run Envoy
as `Network Configuration Operators` user.

This patch does the following:

1. Creates user `EnvoyUser` in the container
2. Adds the user to the group [Network Configuration Operators](https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/security-identifiers-in-windows#sids-added-by-windows-server-2003-and-later-versions)

Note that we do not envoy as this user but with this addition any operator has the option to do so without altering the image.

Signed-off-by: Sotiris Nanopoulos <sonanopo@microsoft.com>

Additional Description: N/A
Risk Level: Low
Testing: manual
Docs Changes: N/A
Release Notes: windows container image: added user, EnvoyUser which is part of the Network Configuration Operators group to the container image.
Platform Specific Features: Windows Docker Image

### CODE REVIEW

**Comment by davinci26** [comment]
cc @envoyproxy/windows-dev

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -95,7 +95,7 @@ New Features
 * thrift_proxy: added per upstream metrics within the :ref:`thrift router <envoy_v3_api_msg_extensions.filters.network.thrift_proxy.router.v3.Router>` for request and response size histograms.
 * tls: allow dual ECDSA/RSA certs via SDS. Previously, SDS only supported a single certificate per context, and dual cert was only supported via non-SDS.
 * udp_proxy: added :ref:`key <envoy_v3_api_msg_extensions.filters.udp.udp_proxy.v3.UdpProxyConfig.HashPolicy>` as another hash policy to support hash based routing on any given key.
-
+* windows container image: added user, EnvoyUser which is part of the Network Configuration Operators group to the container image.
 Deprecated
```
Comment (wrowe) [inline docs/root/version_history/current.rst:99]: Missing newline?

**Review by wrowe — APPROVED** [review]
Otherwise LGTM, only wondering if there are any service management/use-as-service-account aspects we should consider, and what run-as-service account we should be using by example in our docs.

File: docs/root/version_history/current.rst (hunk 1 above)
Comment (davinci26) [reply docs/root/version_history/current.rst:100]: fixed

**Comment by davinci26** [comment]
> if there are any service management/use-as-service-account aspects we should consider

I think that from all the known security groups this one is the one that fits Envoy's description more accurately. We went internally thought that exercise and this is the best group we could see fit.

Down the line this will give us more flexibility with calling various win32 `ioctl` which require elevated network privilege.

**Review by wrowe — APPROVED** [review]

**Pull request merged**